annotated_types/__init__.py

```diff
@@ -239,6 +261,13 @@ class Len(BaseMetadata):
     min_inclusive: Annotated[int, Ge(0)] = 0
     max_exclusive: Optional[Annotated[int, Ge(0)]] = None
 
+    def __iter__(self) -> Iterator[BaseMetadata]:
+        """Unpack a Len into zone or more single-bounds."""
+        if self.min_inclusive > 0:
```

Comment:
only question is whether we should always yield `MinLen`, but I'd prefer to skip the constraint if it's unnecessary.